Packs/MicrosoftGraphMail/Integrations/MicrosoftGraphMail/MicrosoftGraphMail.py

```diff
@@ -1150,14 +1152,48 @@ def delete_mail_command(client: MsGraphClient, args):
     return_outputs(human_readable, entry_context)
 
 
+def item_result_creator(raw_response, user_id):
+    item = raw_response.get('item', {})
+    item_type = item.get('@odata.type', '')
+    if 'message' in item_type:
+        message_id = raw_response.get('id')
+        item['id'] = message_id
+        mail_context = build_mail_object(item, user_id=user_id, get_body=True)
+        human_readable = tableToMarkdown(
+            f'Attachment ID {message_id} \n **message details:**',
+            mail_context,
+            headers=['ID', 'Subject', 'SendTime', 'Sender', 'From', 'HasAttachments', 'Body']
+        )
+        return CommandResults(outputs_prefix='MSGraphMail',
```

Comment:
I second glance, note that you should not call CommandResults from the internal function, but rather from the command function, to make this more general business logic function.
also, it is better to create the human readable out there if possible.

Reply:
It is not completely different, it try to do the same but returning eml file type. From my testing it does the same in fetch incident and in get-attachment, maybe we want to get the option to the customer to get attachment by eml type